chainer/datasets/sub_dataset.py

```diff
@@ -0,0 +1,157 @@
+import numpy
+import six
+
+from chainer.dataset import dataset_mixin
+
+
+class SubDataset(dataset_mixin.DatasetMixin):
+
+    """Subset of a base dataset.
+
+    SubDataset defines a subset of a given base dataset. The subset is defined
+    as an interval of indexes, optionally with a given permutation.
+
+    If ``order`` is given, then the ``i``-th example of this dataset is the
+    ``order[start + i]``-th example of the base dataset, where ``i`` is a
+    non-negative integer. If ``order`` is not given, then the ``i``-th example
+    of this dataset is the ``start + i``-th example of the base dataset.
+    Negative indexing is also allowed: in this case, the term ``start + i`` is
+    replaced by ``finish + i``.
+
+    Args:
+        dataset: Base dataset.
```

Comment:
OK, thank you for clarification. I thought we should write the class of `dataset`. But "a derived class of `DatasetMixin`" is somewhat unnatural. Do you have any idea?